src/subcommand/wallet/inscribe.rs

```diff
@@ -0,0 +1,141 @@
+use super::*;
+
+use bitcoin::{
+  blockdata::{opcodes, script},
+  secp256k1::{self, rand, KeyPair, Secp256k1, XOnlyPublicKey},
+  util::sighash::{Prevouts, SighashCache},
+  util::taproot::TaprootBuilder,
+  util::taproot::{LeafVersion, TapLeafHash},
+  PackedLockTime, SchnorrSighashType, Witness,
+};
+
+#[derive(Debug, Parser)]
+pub(crate) struct Inscribe {
+  ordinal: Ordinal,
+  content: String,
+}
+
+impl Inscribe {
+  pub(crate) fn run(self, options: Options) -> Result {
+    let client = options.bitcoin_rpc_client_mainnet_forbidden("ord wallet inscribe")?;
+
+    let index = Index::open(&options)?;
+
+    index.update()?;
+
+    let secp256k1 = Secp256k1::new();
+    let key_pair = KeyPair::new(&secp256k1, &mut rand::thread_rng());
+    let (public_key, _parity) = XOnlyPublicKey::from_keypair(&key_pair);
+
+    let script = script::Builder::new()
+      .push_slice(&public_key.serialize())
+      .push_opcode(opcodes::all::OP_CHECKSIG)
+      .push_opcode(opcodes::OP_FALSE)
+      .push_opcode(opcodes::all::OP_IF)
+      .push_slice(self.content.as_bytes())
+      .push_opcode(opcodes::all::OP_ENDIF)
+      .into_script();
+
+    let taproot_spend_info = TaprootBuilder::new()
+      .add_leaf(0, script.clone())
+      .unwrap()
+      .finalize(&secp256k1, public_key)
+      .unwrap();
+
+    let address = Address::p2tr_tweaked(taproot_spend_info.output_key(), options.chain.network());
+
+    let utxos = list_unspent(&options, &index)?;
+
+    let change = vec![
+      client
```

Comment:
Deduplicate these calls. We could have a `fn get_raw_change_addresses(client: &Client, n: usize) -> Vec<Address>`. I think we maybe also use this in a few other places.